src/sdk/wallet.js

```diff
@@ -0,0 +1,192 @@
+            throw new Error('Pre-approval transaction failed!');
+
+        try {
+            var depositTx = await clientFund.depositTokens(tokenContract.address, amountBN, options);
+            dbg('DepositTX: ' + JSON.stringify(depositTx));
+
+            var depositTxReceipt = await getTransactionReceipt.call(this, depositTx.hash);
+            dbg('DepositTX (receipt): ' + JSON.stringify(depositTxReceipt));
+        }
+        catch (err) {
+            dbg(err);
+            throw new Error('Failed to deposit token in time!')
+        }
+
+        if (depositTxReceipt.status === 0)
+            throw new Error('Deposit transaction failed!');
+
+        return [approvalTxReceipt, depositTxReceipt];
+    }
+}
+
+module.exports = Wallet;
+
+/**
+ * Private method - invoke bound to instance.
+ * Attempts to get transaction receipt once every second for up to 1 minute.
+ * @see https://docs.ethers.io/ethers.js/html/api-providers.html#transaction-receipts
+ * @param {string} transactionHash
+ * @returns {Promise<Object>}
+ */
+function getTransactionReceipt(transactionHash) {
+    let retries = 0;
+    return Promise.retry(() => {
+        process.stderr.write('.');
+        return deferredIsNotNull(this.provider.getTransactionReceipt(transactionHash))
+    }, 60, 1000).then(result => {
+        process.stderr.write('\n');
+        return result;
+    });
+}
+
+/**
+ * Private method - invoke bound to instance.
+ * Retrieves information about the token that has the specified symbol.
+ * @param {string} symbol
+ * @returns {Promise<Object>}
+ */
+async function getTokenInfo(symbol) {
+    const supportedTokens = await this.provider.getSupportedTokens();
+    const tokenInfo = supportedTokens.find(t => t.symbol.toUpperCase() === symbol.toUpperCase());
+    if (!tokenInfo)
+        throw new Error('Unknown currency. See "striim show tokens" for a list of supported tokens.');
+    return tokenInfo;
+}
+
+/**
+ * Private method - invoke bound to instance.
+ * Retrieves the API access token for striim APIs.
+ * @returns {Promise<void>}
+ */
+async function getApiAccessToken() {
+    return this.provider.getApiAccessToken();
+}
+
+Promise.retry = function(attemptFn, times, delay) {
+    return new Promise(function(resolve, reject) {
+        let error;
+
+        function attempt() {
+            if (!times)
+                return reject(error);
+
+            attemptFn()
+                .then(resolve)
+                .catch(function(e) {
+                    times--;
+                    error = e;
+                    setTimeout(function() {
+                        attempt()
+                    }, delay);
+                });
+        }
+
+        attempt();
+    });
+};
+
+function deferredIsNotNull(promise) {
+    return new Promise((resolve, reject) => {
+        promise.then(res => {
+            if (res !== null)
+                resolve(res);
+            else
+                reject();
+        })
+    });
+}
+
+function dbg(...args) {
```

Comment:
This is a good utility function, a long with the one you have for logging out errors. Maybe extract to a helper class at some point?